### PR TITLE
solution to parition problem

### PR DESCRIPTION
**Pull request by KamillaKhabibrakhmanova**

### CODE REVIEW

File: Chapter2-LinkedLists/2.4-PartitionLL.js [hunk 1]
```diff
@@ -3,6 +3,26 @@ Problem:  Write code to partition a linked list around a value x, such that all
...
+@params{x} value to partition around
+*/
+
+function partitionLinkedList(firstElement, x) {
+	var prev = null;
+	var curr = firstElement;
+	var headNode = firstElement;
+
+	//if val is less than the partition, set it as the head to move it left
+	while ( prev.next ) {
```
Comment (seemaullal) [inline Chapter2-LinkedLists/2.4-PartitionLL.js:16]: since prev is set to `null` on line 12, it looks like this `while` loop will never be entered

**Comment by KamillaKhabibrakhmanova** [comment]
fixed!

**Comment by seemaullal** [comment]
I believe this results in an infinite loop. Tried testing this but it just kept going. Adding some comments to the code as well. You can see the test I did here: https://repl.it/C3zB/4

**Comment by KamillaKhabibrakhmanova** [comment]
this one should _finally_ work: https://repl.it/C3zB/5

**Comment by seemaullal** [comment]
@KamillaKhabibrakhmanova Looks good! Maybe we can write some tests later, but thanks for making the changes :) Are you able to merge this or should I do that? :shipit:

**Pull request merged**